sdk/internal/recording/recording.go

```diff
@@ -577,6 +579,105 @@ func getTestId(pathToRecordings string, t *testing.T) string {
 	return path.Join(pathToRecordings, "recordings", t.Name()+".json")
 }
 
+func getGitRoot(fromPath string) (string, error) {
```

Comment:
The better way to do this is to execute the following command.
```
git rev-parse --show-toplevel
```
This will return the repo root.
```
PS C:\git\Azure\azure-sdk-for-go\sdk\resourcemanager\compute\armcompute> git rev-parse --show-toplevel
C:/git/Azure/azure-sdk-for-go
```

Reply:
Fair point, if we're ok shelling out to git in the unit tests (I don't see an issue with that).

Reply:
I have no issues with it and have used this in CI in the past.  Looks like we also use this same technique in the [smoketest tool](https://github.com/Azure/azure-sdk-for-go/blob/main/eng/tools/smoketests/cmd/root.go#L176-L180).

Reply:
Updated, much simpler now. Thanks!